apps/dashboard/src/components/workflow-editor/schema.ts

```diff
@@ -61,13 +61,15 @@ export const workflowSchema = workflowMinimalSchema.extend({
      * TODO: Add user schema
      */
     user: z.any().nullable(),
-    default: z.object({
-      all: z.object({
-        enabled: z.boolean(),
-        readOnly: z.boolean(),
-      }),
-      channels: channelsSchema,
-    }),
+    default: z
+      .object({
+        all: z.object({
+          enabled: z.boolean(),
+          readOnly: z.boolean(),
+        }),
+        channels: channelsSchema,
+      })
+      .nullable(),
```

Comment:
The `default` preferences isn't nullable, it should always be present. The `default` preferences for Dashboard-sourced workflows will always be the `DEFAULT_WORKFLOW_PREFERENCES`, whereas the `default` preferences for Framework-sourced workflows are that defined in code.

What issue is this solving?

Reply:
If no `default` preferences are supplied, the `UpsertWorkflow` use-case falls back to `DEFAULT_WORKFLOW_PREFERENCES` here. So instead, we should not send `default`.